File: python/anagram/anagram.py
# from collections import Counter as multiset


def find_anagrams(word, candidates):
    return [
        item
        for item in candidates
        if (
            sorted(item.lower()) == sorted(word.lower())
            and item.lower() != word.lower()
        )
    ]
    # return [item for item in candidates if (multiset(item.lower()) == multiset(word.lower()) and and item.lower() != word.lower())]


def version():
    VERSION = 1

File: go/anagram/anagram.go
package anagram

import (
	"sort"
	"strings"
)

const version = 1

// alphagram returns the chars sorted of a word.
func alphagram(subject string) string {
	chars := strings.Split(strings.ToLower(subject), "")
	sort.Strings(chars)
	return strings.Join(chars, "")
}

func Detect(subject string, candidates []string) (matches []string) {
	alphaWord := alphagram(subject)
	for _, candidate := range candidates {
		if alphaWord == alphagram(candidate) && !strings.EqualFold(subject, candidate) {
			matches = append(matches, candidate)
		}
	}
	return
}
